gems/aws-sdk-core/lib/aws-sdk-core/sso_credentials.rb

```diff
@@ -100,7 +100,7 @@ def read_cached_token
         raise ArgumentError, 'Cached SSO Token is expired.'
       end
       cached_token
-    rescue Aws::Json::ParseError, ArgumentError
+    rescue Errno::ENOENT, Aws::Json::ParseError, ArgumentError
```

Comment:
Probably best to use `NETWORK_ERRORS` from other credential providers, and perhaps two layers of rescue?

Reply:
This doesn't overlap at all with the [NETWORK_ERRORS](https://github.com/aws/aws-sdk-ruby/blob/version-3/gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb#L25) - this is purely a local issue when the file is not found locally.  

What do we get with two layers of rescue?

Reply:
Oh duh.. this is cache file miss. 🤦